src/Particles/babylon.particleSystem.ts

```diff
@@ -540,10 +561,144 @@
             this._vertexData[offset + 11] = particle.cellIndex;
         }
 
+        // start of sub system methods
+
+        /**
+         * "Recycles" one of the particle by copying it back to the "stock" of particles and removing it from the active list.
+         * Its lifetime will start back at 0.
+         * @param particle The particle to recycle
+         */
+        public recycleParticle: (particle: Particle) => void = (particle) => {
+            ParticleSystem.recycleParticle(this, this, particle);
+        };
+
+        private _stopSubEmitters(): void {
+            this.stockSubSystems.forEach(generation => {
+                this.stockSubSystems.get(generation)!.forEach(subSystem => {
+                    subSystem.stop();
+                });
+            });
+
+            this.activeSubSystems.forEach(subSystemName => {
+                var subSystem = this.activeSubSystems.get(subSystemName)!;
+                subSystem.stop();
+                subSystem._stoppedEmitting(true);
+            });
+            this.activeSubSystems = new StringDictionary<ParticleSystem>();
+        }
+
+        private static emitFromGeneration(rootSystem: ParticleSystem, particle: Particle, generation: number): void {

```

Comment:
not sure why you will need a generation mechanism.
If a particle dies and the root particle system has sub emitters then spawn a new system and run it.

What else am I missing?

Reply:
If I'm right (and I'm probably wrong :)) then we don't need all the static functions

Reply:
We don't need the static that's correct I did this way to force not using this because it was too confusing :) but we need the generation to know which sub emit we are currently at hence we can decide to stop or continue.

I will remove the static :)